src/test/rustdoc/macro_rules-matchers.rs

```diff
@@ -0,0 +1,41 @@
+// This is a regression test for issue #86208.
+// It is also a general test of macro_rules! display.
+
+#![crate_name = "foo"]
+
+// @has 'foo/macro.todo.html'
+// @has - '//span[@class="macro"]' 'macro_rules!'
+// @has - '//span[@class="ident"]' 'todo'
+// Note: count = 2 * ('=' + '>') + '+' = 2 * (1 + 1) + 1 = 5
+// @count - '//pre[@class="rust macro"]//span[@class="op"]' 5
+
+// @has - '{ ()'
+// @has - '//span[@class="op"]' '='
+// @has - '//span[@class="op"]' '>'
+// @has - '{ ... };'
+
+// @has - '($ ($'
+// @has - '//span[@class="ident"]' 'arg'
+// @has - ':'
+// @has - '//span[@class="ident"]' 'tt'
+// @has - '//span[@class="op"]' '+'
+// @has - ')'
```

Comment:
Do you know of a way to do a `@has` check that matches on the `innerText` of an element (in this case, `<pre class="rust macro">`) to avoid this mess?

Reply:
I don't, @GuillaumeGomez may.

Actually my understanding is that is *always* used innerText, are you sure doing it the obvious way doesn't work?

Reply:
Just like @jyn514 said. The problem might be that the text doesn't "look like" what you might expect but that's it. :)

Reply:
I had tried something like `@has - 'macro_rules! todo {'` and it failed; maybe I need to explicitly specify `//pre`? I also had this issue in #83501 where `@has - 'Size: 1 byte'` didn't work because `Size:` was wrapped with `<b>`.

Reply:
Ah! Maybe try with `//text()` at the end.

Reply:
Hmm, it doesn't seem to be working. `//text()` by itself gave this error:

```
27: Non-absolute XPath is not supported due to implementation issues
	    // @has - '//text()' 'macro_rules! macro1 {'
28: Non-absolute XPath is not supported due to implementation issues
	    // @has - '//text()' '() => { ... };'
29: Non-absolute XPath is not supported due to implementation issues
	    // @has - '//text()' '$ ($ arg : expr), +) => { ... };'
30: Non-absolute XPath is not supported due to implementation issues
	    // @has - '//text()' '}'

Encountered 4 errors
```

and `//pre//text()` gave this:

```
27: @has check failed
	`XPATH PATTERN` did not match
	    // @has - '//pre//text()' 'macro_rules! macro1 {'
28: @has check failed
	`XPATH PATTERN` did not match
	    // @has - '//pre//text()' '() => { ... };'
29: @has check failed
	`XPATH PATTERN` did not match
	    // @has - '//pre//text()' '$ ($ arg : expr), +) => { ... };'
30: @has check failed
	`XPATH PATTERN` did not match
	    // @has - '//pre//text()' '}'

Encountered 4 errors
```

I also tried `//pre///text()` (extra slash), but that didn't work either.